### PR TITLE
Allow to customize butane config file

### PR DESCRIPTION
**Pull request by tobiashuste**
This way we can support other operating systems than Flatcar Linux as well, e.g. Fedora CoreOS.

### CODE REVIEW

File: roles/gitlab_runner/defaults/main.yml [hunk 1]
```diff
@@ -49,3 +49,5 @@ gitlab_runner_session_server_timeout: 1800
 gitlab_runner_autoscaler_plugin_version: "0.21.1"
 gitlab_runner_autoscaler_plugin_url: "https://github.com/sardinasystems/fleeting-plugin-openstack/releases/download/{{ gitlab_runner_autoscaler_plugin_version }}/fleeting-plugin-openstack_{{ gitlab_runner_autoscaler_plugin_version }}_linux_amd64.tar.gz"
 gitlab_runner_autoscaler_plugin_checksumfile: "https://github.com/sardinasystems/fleeting-plugin-openstack/releases/download/{{ gitlab_runner_autoscaler_plugin_version }}/fleeting-plugin-openstack_{{ gitlab_runner_autoscaler_plugin_version }}_sha512-checksums.txt"
+
+gitlab_runner_butane_config: "butane-config.bu.j2"
```
Comment (christianhueserhzdr) [inline roles/gitlab_runner/defaults/main.yml:53]: Could it be also `butane-config.bu` without the template fileending and use that one throughout the role?

```suggestion
gitlab_runner_butane_config: "butane-config.bu"
```

File: roles/gitlab_runner/handlers/main.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@
 ---
 
 - name: "Transpile the flatcar linux configuration"
-  ansible.builtin.command: "butane -o /etc/gitlab-runner/ignition.json /etc/gitlab-runner/flatcar-linux-config.bu"
+  ansible.builtin.command: "butane -o /etc/gitlab-runner/ignition.json /etc/gitlab-runner/butane-config.bu"
```
Comment (christianhueserhzdr) [inline roles/gitlab_runner/handlers/main.yml:9]:
```suggestion
  ansible.builtin.command: "butane -o /etc/gitlab-runner/ignition.json /etc/gitlab-runner/{{ gitlab_runner_butane_config }}"
```

File: roles/gitlab_runner/tasks/configuration.yml [hunk 3]
```diff
@@ -41,8 +41,8 @@
 
 - name: "Place the container linux configuration on the host"
   ansible.builtin.template:
-    src: "flatcar-linux-config.bu.j2"
-    dest: "/etc/gitlab-runner/flatcar-linux-config.bu"
+    src: "{{ gitlab_runner_butane_config }}"
```
Comment (christianhueserhzdr) [inline roles/gitlab_runner/tasks/configuration.yml:44]:
```suggestion
    src: "{{ gitlab_runner_butane_config }}.j2"
```

File: roles/gitlab_runner/tasks/configuration.yml [hunk 4]
```diff
@@ -41,8 +41,8 @@
...
+    dest: "/etc/gitlab-runner/butane-config.bu"
```
Comment (christianhueserhzdr) [inline roles/gitlab_runner/tasks/configuration.yml:45]:
```suggestion
    dest: "/etc/gitlab-runner/{{ gitlab_runner_butane_config }}"
```

File: roles/gitlab_runner/tasks/configuration.yml [hunk 5]
```diff
@@ -75,7 +75,7 @@
       check_mode: false
 
     - name: "Dry-run of transpile the flatcar linux configuration"
-      ansible.builtin.command: "butane -o {{ (temp_directory.path, 'ignition.json') | path_join }} /etc/gitlab-runner/flatcar-linux-config.bu"
+      ansible.builtin.command: "butane -o {{ (temp_directory.path, 'ignition.json') | path_join }} /etc/gitlab-runner/butane-config.bu"
```
Comment (christianhueserhzdr) [inline roles/gitlab_runner/tasks/configuration.yml:78]:
```suggestion
      ansible.builtin.command: "butane -o {{ (temp_directory.path, 'ignition.json') | path_join }} /etc/gitlab-runner/{{ gitlab_runner_butane_config }}"
```

**Review by christianhueserhzdr — COMMENTED** [review]
Thank you very much for making the butane config file customizable. I would do it the other way round and introduce the butane config file name as a variable instead of the template file name. Could this also be a way to go?

**Comment by tobiashuste** [comment]
> Thank you very much for making the butane config file customizable. I would do it the other way round and introduce the butane config file name as a variable instead of the template file name. Could this also be a way to go?

This change was intended to allow for supporting more than only Flatcar Linux as Operating System. We could als support https://fedoraproject.org/coreos/ with this change. Thus, the template was important to be changeable. The server side file does not matter.
I suggest to give the variable a better name: `gitlab_runner_butane_config_template`

**Review by christianhueserhzdr — APPROVED** [review]

**Pull request merged**